CI: llvm@19 bottle installation fails with incorrect path lookup

### What were you trying to do (and why)?

I was trying to submit a PR (https://github.com/Homebrew/homebrew-core/pull/234822) to fix typos in the llgo formula test strings,  changing "LLGO" to "LLGo" for consistency. This is a minor documentation/test fix that should not affect the actual functionality. A maintainer advised me to create an issue for this CI failure  (https://github.com/Homebrew/homebrew-core/pull/234822#issuecomment-3219784075).

### `brew gist-logs <formula>` link OR `brew config` AND `brew doctor` output

```shell
homebrew-core on  llgo/typo via 💎 v2.6.10 
❯ brew config
HOMEBREW_VERSION: 4.6.4-54-g99cdd7d
ORIGIN: https://mirrors.tuna.tsinghua.edu.cn/git/homebrew/brew.git
HEAD: 99cdd7d8c06c92bb28b97a8f0855cdc0b89fb5a3
Last commit: 6 days ago
Branch: main
Core tap HEAD: 6cbf096b7666bea176d2d18a003c5951667dbe73
Core tap last commit: 9 weeks ago
Core tap JSON: 31 Jul 14:37 UTC
Core cask tap JSON: 31 Jul 14:37 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_FORBID_PACKAGES_FROM_PATHS: set
HOMEBREW_MAKE_JOBS: 8
HOMEBREW_SORBET_RUNTIME: set
Homebrew Ruby: 3.4.5 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.4.5/bin/ruby
CPU: octa-core 64-bit arm_ibiza
Clang: 17.0.0 build 1700
Git: 2.39.5 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 8.7.1 => /usr/bin/curl
macOS: 15.5-arm64
CLT: 16.4.0.0.1.1747106510
Xcode: 16.4
Rosetta 2: false


homebrew-core on  llgo/typo via 💎 v2.6.10 
❯ brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Suspicious https://github.com/Homebrew/brew git origin remote found.
The current git origin is:
  https://mirrors.tuna.tsinghua.edu.cn/git/homebrew/brew.git

With a non-standard origin, Homebrew won't update properly.
You can solve this by setting the origin remote:
  git -C "/opt/homebrew" remote set-url origin https://github.com/Homebrew/brew

Warning: Suspicious https://github.com/Homebrew/homebrew-core git origin remote found.
The current git origin is:
  https://mirrors.tuna.tsinghua.edu.cn/git/homebrew/homebrew-core.git

With a non-standard origin, Homebrew won't update properly.
You can solve this by setting the origin remote:
  git -C "/opt/homebrew/Library/Taps/homebrew/homebrew-core" remote set-url origin https://github.com/Homebrew/homebrew-core

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  icu4c@76
  tinyxml

Warning: You have the following deprecated, official taps tapped:
  Homebrew/homebrew-services
Untap them with `brew untap`.

Warning: Unbrewed dylibs were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae and may need to be deleted.

Unexpected dylibs:
  /usr/local/lib/libcsv_wrapper.dylib
  /usr/local/lib/libhyper.dylib
  /usr/local/lib/libpstr.dylib
  /usr/local/lib/libsled.dylib

Warning: Unbrewed header files were found in /usr/local/include.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae and may need to be deleted.

Unexpected header files:
  /usr/local/include/sled/*

Warning: Unbrewed static libraries were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae and may need to be deleted.

Unexpected static libraries:
  /usr/local/lib/libsled.a

Warning: You have uncommitted modifications to Homebrew/homebrew-core.
If this is a surprise to you, then you should stash these modifications.
Stashing returns Homebrew to a pristine state but can be undone
should you later need to do so for some reason.
  git -C "/opt/homebrew/Library/Taps/homebrew/homebrew-core" stash -u && git -C "${path}" clean -d -f

Uncommitted files:
   M Formula/l/llgo.rb

Warning: You have pkgconf installed that was built on macOS 14
         but you are running macOS 15.

This can cause issues with packages that depend on system libraries, such as libffi.
To fix this issue, reinstall pkgconf:
  brew reinstall pkgconf

For more information, see: https://github.com/Homebrew/brew/issues/16137
We'd welcome a PR to automatically mitigate this instead of just warning about it.

Warning: Some taps are not on the default git origin branch and may not receive
updates. If this is a surprise to you, check out the default branch with:
  git -C $(brew --repo homebrew/core) checkout main
(base)
```

### Verification

- [x] My `brew doctor` output says `Your system is ready to brew.` and am still able to reproduce my issue.
- [x] I ran `brew update` and am still able to reproduce my issue.
- [x] I have resolved all warnings from `brew doctor` and that did not fix my problem.
- [x] I searched for recent similar issues at https://github.com/Homebrew/homebrew-core/issues?q=is%3Aissue and found no duplicates.
- [x] My issue is not about a failure to build a formula from source.



### What happened (include all command output)?

The macOS 14-arm64 CI build is consistently failing during bottle dependency installation with the following error:
```
Run xargs -t brew test-bot \
+ xargs -t brew test-bot --testing-formulae=llgo --added-formulae= --deleted-formulae=
brew test-bot --testing-formulae=llgo --added-formulae= --deleted-formulae= --only-formulae --junit --only-json-tab --skip-dependents --concurrent-downloads
==> Using Homebrew/homebrew-test-bot e8ea2be (Merge pull request Homebrew/homebrew-core#1547 from Homebrew/utils_output)
==> Using Homebrew/brew 4.6.7-13-gdd7e3e0142 (Merge pull request Homebrew/homebrew-core#20566 from Homebrew/extract-pathname-refinement)
==> Testing Homebrew/homebrew-core 92197687efd (Merge 53bbac8ecd3366eb6f31d66d546130e1b43bbeee into f6344169ce391364d53649777e1f24b5309131ec):

==> Running Formulae#run!

==> Running Formulae#formula!(llgo)
==> brew deps --tree --annotate --include-build --include-test llgo
==> Determining dependencies...
==> brew fetch --formulae --retry autoconf automake bdw-gc ca-certificates certifi cmake expat go go@1.24 libffi libtool libuv lld@19 llvm@19 lz4 lzip m4 mpdecimal ninja openssl@3 pcre2 pkgconf python@3.12 python@3.13 readline sphinx-doc sqlite swig xz zstd
==> brew install --only-dependencies --verbose --formula --build-bottle llgo
==> Starting tests for llgo
==> brew fetch --formula --retry llgo --build-bottle --force
==> brew install --verbose --formula --build-bottle llgo
==> brew livecheck --autobump --formula --json --full-name llgo
==> brew style --formula llgo
==> brew audit --formula llgo --online --git --skip-style
==> brew bottle --verbose --json llgo --only-json-tab
==> brew bottle --merge --write --no-commit --no-all-checks ./llgo--0.11.5.arm64_sonoma.bottle.json
==> brew uninstall --formula --force --ignore-dependencies llgo
==> brew uninstall --formulae --force --ignore-dependencies autoconf automake certifi cmake expat go libtool lzip m4 mpdecimal ninja pcre2 python@3.12 python@3.13 readline sphinx-doc sqlite swig
==> brew install --only-dependencies ./llgo--0.11.5.arm64_sonoma.bottle.2.tar.gz
==> FAILED
Full install --only-dependencies ./llgo--0.11.5.arm64_sonoma.bottle.2.tar.gz output
  ==> Fetching downloads for: llgo
  ✔︎ Bottle Manifest bdw-gc (8.2.8)
  ✔︎ Bottle bdw-gc (8.2.8)
  ✔︎ Bottle Manifest go@1.24 (1.24.6)
  ✔︎ Bottle Manifest libffi (3.5.2)
  ✔︎ Bottle libffi (3.5.2)
  ✔︎ Bottle Manifest libuv (1.51.0)
  ✔︎ Bottle libuv (1.51.0)
  ✔︎ Bottle Manifest xz (5.8.1)
  ✔︎ Bottle xz (5.8.1)
  ✔︎ Bottle Manifest lz4 (1.10.0)
  ✔︎ Bottle lz4 (1.10.0)
  ✔︎ Bottle Manifest zstd (1.5.7)
  ✔︎ Bottle zstd (1.5.7)
  ✔︎ Bottle Manifest llvm@19 (19.1.7)
  ✔︎ Bottle Manifest lld@19 (19.1.7)
  ✔︎ Bottle Manifest ca-certificates (2025-08-12)
  ✔︎ Bottle ca-certificates (2025-08-12)
  ✔︎ Bottle Manifest openssl@3 (3.5.2)
  ✔︎ Bottle Manifest pkgconf (2.5.1)
  ✔︎ Bottle lld@19 (19.1.7)
  ✔︎ Bottle lld@19 (19.1.7)
  ✔︎ Bottle pkgconf (2.5.1)
  ✔︎ Bottle openssl@3 (3.5.2)
  ✔︎ Bottle go@1.24 (1.24.6)
  ✔︎ Bottle llvm@19 (19.1.7)
  ==> Installing dependencies for llgo: bdw-gc, go@1.24, libffi, libuv, xz, lz4, zstd, llvm@19, lld@19, ca-certificates, openssl@3 and pkgconf
  ==> Installing llgo dependency: bdw-gc
  ==> Pouring bdw-gc--8.2.8.arm64_sonoma.bottle.tar.gz
  /opt/homebrew/Cellar/bdw-gc/8.2.8: 82 files, 1.8MB
  ==> Installing llgo dependency: go@1.24
  ==> Pouring go@1.24--1.24.6.arm64_sonoma.bottle.tar.gz
  /opt/homebrew/Cellar/go@1.24/1.24.6: 14,120 files, 243.6MB
  ==> Installing llgo dependency: libffi
  ==> Pouring libffi--3.5.2.arm64_sonoma.bottle.tar.gz
  /opt/homebrew/Cellar/libffi/3.5.2: 18 files, 812.5KB
  ==> Installing llgo dependency: libuv
  ==> Pouring libuv--1.51.0.arm64_sonoma.bottle.tar.gz
  /opt/homebrew/Cellar/libuv/1.51.0: 20 files, 1.2MB
  ==> Installing llgo dependency: xz
  ==> Pouring xz--5.8.1.arm64_sonoma.bottle.tar.gz
  /opt/homebrew/Cellar/xz/5.8.1: 96 files, 2.5MB
  ==> Installing llgo dependency: lz4
  ==> Pouring lz4--1.10.0.arm64_sonoma.bottle.1.tar.gz
  /opt/homebrew/Cellar/lz4/1.10.0: 24 files, 714.2KB
  ==> Installing llgo dependency: zstd
  ==> Pouring zstd--1.5.7.arm64_sonoma.bottle.1.tar.gz
  /opt/homebrew/Cellar/zstd/1.5.7: 32 files, 2.2MB
  ==> Installing llgo dependency: llvm@19
  ==> Pouring llvm@19--19.1.7.arm64_sonoma.bottle.tar.gz
  Error: No such file or directory @ rb_file_s_lstat - /opt/homebrew/Cellar/llvm/19.1.7
==> brew install ./llgo--0.11.5.arm64_sonoma.bottle.2.tar.gz
==> brew linkage --test llgo
==> brew linkage --cached --test --strict llgo
==> brew linkage --cached llgo
==> brew install --formula --only-dependencies --include-test llgo
==> brew test --verbose llgo
==> brew uninstall --formulae --force --ignore-dependencies bdw-gc ca-certificates go@1.24 libffi libuv lld@19 llvm@19 lz4 openssl@3 pkgconf xz zstd

Error: 1 failed step!
brew install --only-dependencies ./llgo--0.11.5.arm64_sonoma.bottle.2.tar.gz
Error: Process completed with exit code 1.
```

[logs_44130990797 (1).zip](https://github.com/user-attachments/files/21970049/logs_44130990797.1.zip)

### What did you expect to happen?

expect updated success 

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
with the llgo.rb from this pr https://github.com/Homebrew/homebrew-core/pull/234822
```

--- a/Library/Homebrew/formula_cellar_checks.rb
+++ b/Library/Homebrew/formula_cellar_checks.rb
@@ -373,7 +373,7 @@ module FormulaCellarChecks
     universal_binaries_expected = if (formula_tap = formula.tap).present? && formula_tap.core_tap?
       formula_name = formula.name
       # Apply audit exception to versioned formulae too from the unversioned name.
-      formula_name.gsub!(/@\d+(\.\d+)*$/, "") if formula.versioned_formula?
+      formula_name = formula_name.gsub(/@\d+(\.\d+)*$/, "") if formula.versioned_formula?
       formula_tap.audit_exception(:universal_binary_allowlist, formula_name)
     else
       true

